src/MultiTypeSymbolTable.chpl

```diff
@@ -199,32 +176,17 @@ module MultiTypeSymbolTable
         :returns: bool indicating whether the deletion occurred
         */
         proc deleteEntry(name: string): bool throws {
-            if tab.contains(name) { 
-               if !registry.contains(name) {
-                   mtLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                           "Deleting unregistered entry: %s".format(name)); 
-                   tab.remove(name);
-                   return true;
-               } else {
-                    mtLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                           "Skipping registered entry: %s".format(name)); 
-                    return false;
-               }            
+            checkTable(name, "deleteEntry");
+            if !registry.contains(name) {
+                mtLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                       "Deleting unregistered entry: %s".format(name)); 
+                tab.remove(name);
+                return true;
             } else {
-                if registry.contains(name) {
-                    mtLogger.error(getModuleName(),getRoutineName(),getLineNumber(),
-                                     "Registered entry is not in SymTab: %s".format(name));
-                } else {
```

Comment:
This condition in `deleteEntry` is one area that I was unsure about. This case of "registered object that's not in the symbol table" is no longer captured and logged since the generic "in symbol table check" is done before checking in registry. I could switch the order and add the logging back in if people want. My thinking was this shouldn't happen and the code is cleaner without multiple cases

Reply:
Just to make sure I understand this correctly:  In both implementations if the item is not in the SymTab we're still going to throw an error.  So the difference here is whether we log additional information about the missing item (whether it was registered or not).

So my (rhetorical) question is what would we do with that additional information.  Is it going to help us determine what happened / what led to the item being missing?  I'm not sure, I think I'm leaning towards no it's not really helpful, but I'm still a bit of a newb here.

If we decided that extra logging information is useful, I'd probably wrap the `checkTable` call in a try/catch, log the registry info and then re-throw the error.